src/NHibernate/Logging.cs

```diff
@@ -498,114 +656,139 @@ public Log4NetLogger(object logger)
-			if (IsErrorEnabled)
-				ErrorExceptionDelegate(_logger, message, exception);
+			switch (logLevel)
+			{
+				case InternalLogLevel.Debug:
+				case InternalLogLevel.Trace:
+					if (exception != null)
+						DebugExceptionDelegate(_logger, state, exception);
+					else if (state.Args?.Length > 0)
+						DebugFormatDelegate(_logger, state.Format, state.Args);
+					else
+						DebugDelegate(_logger, state);
+					break;
+				case InternalLogLevel.Info:
+					if (exception != null)
+						InfoExceptionDelegate(_logger, state, exception);
+					else if (state.Args?.Length > 0)
+						InfoFormatDelegate(_logger, state.Format, state.Args);
+					else
+						InfoDelegate(_logger, state);
+					break;
+				case InternalLogLevel.Warn:
+					if (exception != null)
+						WarnExceptionDelegate(_logger, state, exception);
+					else if (state.Args?.Length > 0)
+						WarnFormatDelegate(_logger, state.Format, state.Args);
+					else
+						WarnDelegate(_logger, state);
+					break;
+				case InternalLogLevel.Error:
+					if (exception != null)
+						ErrorExceptionDelegate(_logger, state, exception);
+					else if (state.Args?.Length > 0)
+						ErrorFormatDelegate(_logger, state.Format, state.Args);
+					else
+						ErrorDelegate(_logger, state);
+					break;
+				case InternalLogLevel.Fatal:
+					if (exception != null)
+						FatalExceptionDelegate(_logger, state, exception);
+					else
+						FatalDelegate(_logger, state);
+					break;
+				case InternalLogLevel.None:
+					break;
+				default:
+					throw new ArgumentOutOfRangeException(nameof(logLevel), logLevel, null);
+			}
 		}
 
-		public void ErrorFormat(string format, params object[] args)
+		/// <inheritdoc />
+		public bool IsEnabled(InternalLogLevel logLevel)
 		{
-			if (IsErrorEnabled)
-				ErrorFormatDelegate(_logger, format, args);
+			switch (logLevel)
+			{
+				case InternalLogLevel.Trace:
+				case InternalLogLevel.Debug:
+					return IsDebugEnabledDelegate(_logger);
+				case InternalLogLevel.Info:
+					return IsInfoEnabledDelegate(_logger);
+				case InternalLogLevel.Warn:
+					return IsWarnEnabledDelegate(_logger);
+				case InternalLogLevel.Error:
+					return IsErrorEnabledDelegate(_logger);
+				case InternalLogLevel.Fatal:
+					return IsFatalEnabledDelegate(_logger);
+				case InternalLogLevel.None:
+					return !IsFatalEnabledDelegate(_logger);
+				default:
+					throw new ArgumentOutOfRangeException(nameof(logLevel), logLevel, null);
+			}
 		}
+		
+		#region IInternalLogger re-implementations required for avoiding a breaking change
+		// If some dependent library use the old internal logger, passing to it objects while using
+		// formatters, we must then implement directly the call. The base impl would instead call
+		// ToString on the object, preventing the formatters to work.
 
-		public void Fatal(object message)
+		void IInternalLogger.Fatal(object message)
 		{
-			if (IsFatalEnabled)
-				FatalDelegate(_logger, message);
+			FatalDelegate(_logger, message);
```

Comment:
Since these methods are obsolete anyway, I do not care about perf here (and log4net does the check itself moreover).